rocksdb_replicator/replicated_db.cpp

```diff
@@ -94,10 +95,11 @@ rocksdb::Status RocksDBReplicator::ReplicatedDB::Write(
   auto end = GetCurrentTimeMs();
   logMetric(kReplicatorWriteMs, start < end ? end - start : 0, db_name_);
   
+  auto replication_mode =  common::DBConfigManager::get()->getReplicationMode(db_name_);
   // for now we have to support both till all clusters are migrated
-  auto replication_mode =  config.replication_mode > FLAGS_replicator_replication_mode ? 
-                            config.replication_mode :  FLAGS_replicator_replication_mode;
-
+  if (FLAGS_replicator_replication_mode > replication_mode) {
```

Comment:
does this mean the global gflag will override the per-dataset config? is that indented?

Reply:
For now yes.. This is because rocksplicator is used by multiple services. Most of them use this flag. Till we migrate all of them , we need to give it priority. Once we migrate all services out of gflags, we will remove this check

Reply:
I see..
this can be very confusing and error-prone, we'll need to distinguish between:
- default ack mode
- per-dataset override
- global gflag override
At least we need to document somewhere in the code on how they work together currently, and how we plan to migrate moving forward.

Reply:
Separately, what do you think of always using `FLAGS_replicator_replication_mode` as a global default for a service, and per-dataset config always override that?

Reply:
@premkumr I still have the above questions, can you clarify the order of applying different overrides to the ack mode

Reply:
@jaricftw oops, I completely skipped this.. Sorry about that ..

> Separately, what do you think of always using FLAGS_replicator_replication_mode as a global default for a service, and per-dataset config always override that?

Technically you are correct. Right now there are only 1/2 datasets that actually use the setting. And we are planning to turn this setting OFF at the gflags and completely control via the config. the default for the gflag is 0  & the default for the config struct is 0 ... So it is going to be controlled by the passed in config...  As we are planning to deprecate this asap, and support the cluster where we are not passing the config file, it would be better to  pick the max for now.. I'll add a TODO.. what do you think ?